libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/core/EventId.kt

```diff
@@ -16,9 +16,18 @@
 
 package io.element.android.libraries.matrix.api.core
 
+import io.element.android.libraries.matrix.api.BuildConfig
 import java.io.Serializable
 
 @JvmInline
 value class EventId(val value: String) : Serializable
 
-fun String.asEventId() = EventId(this)
+fun String.asEventId() = if (MatrixPatterns.isEventId(this)) {
```

Comment:
I don't really like having this extenstion available from every string in the project, can we try to limit it to the module using it? Also are the checks really necessary?

Reply:
It can help to detect issue like [this one](https://github.com/vector-im/element-x-android/pull/318/files#diff-9b8993ef173a404ecbf23a0c01d42502c4e95379978611aa95c3e9fc7785b0a5R72)

Reply:
So I'd do the check only on debug, to avoid wasting resources in release, because we'll get eventId in a lot of places.

Reply:
OK

Reply:
@ganfra 780970414077f833262dbcc1456eb9930c553118